### PR TITLE
feat: VPN Server Configuration resource

### PR DESCRIPTION
**Pull request by ericscheffler**
## Description

This PR introduces a new resource to create VPN Server Configurations within an Azure Virtual WAN, usable by P2S VPN Gateways within a VWAN Virtual Hub. This resource requires a Virtual WAN, but does not require a Virtual Hub to exist before creating the VPN Server Configuration.

Closes Issue [#1333](https://github.com/Azure/Azure-Verified-Modules/issues/1333)

## Pipeline Reference

[avm.res.network.vpn-server-configuration](https://github.com/ericscheffler/bicep-registry-modules/actions/workflows/avm.res.network.vpn-server-configuration.yml)

## Type of Change

- [ ] Update to CI Environment or utilities (Non-module affecting changes)
- [x] Azure Verified Module updates:
  - [ ] Bugfix containing backwards-compatible bug fixes, and I have NOT bumped the MAJOR or MINOR version in `version.json`:
    - [ ] Someone has opened a bug report issue, and I have included "Closes #{bug_report_issue_number}" in the PR description.
    - [ ] The bug was found by the module author, and no one has opened an issue to report it yet.
  - [ ] Feature update backwards compatible feature updates, and I have bumped the MINOR version in `version.json`.
  - [ ] Breaking changes and I have bumped the MAJOR version in `version.json`.
  - [ ] Update to documentation

## Checklist

- [x] I'm sure there are no other open Pull Requests for the same update/change
- [x] I have run `Set-AVMModule` locally to generate the supporting module files.
- [x] My corresponding pipelines / checks run clean and green without any errors or warnings

### CODE REVIEW

File: avm/res/network/vpn-server-configuration/tests/e2e/defaults/main.test.bicep [hunk 1]
```diff
@@ -0,0 +1,84 @@
...
+metadata name = 'Using only defaults'
+metadata description = 'This instance deploys the module with the minimum set of required parameters.'
+
+// ========== //
+// Parameters //
+// ========== //
+
+@description('Optional. The name of the resource group to deploy for testing purposes.')
+@maxLength(90)
+// e.g., for a module 'network/private-endpoint' you could use 'dep-dev-network.privateendpoints-${serviceShort}-rg'
```
Comment (AlexanderSehr) [inline avm/res/network/vpn-server-configuration/tests/e2e/defaults/main.test.bicep:12]: Great work @ericscheffler, just a small ask to please remove these tutorial comments from all test templates :)

File: avm/res/network/vpn-server-configuration/tests/e2e/waf-aligned/main.test.bicep [hunk 2]
```diff
@@ -0,0 +1,87 @@
...
+      vpnAuthenticationTypes: [
+        'AAD'
+      ]
+      vpnProtocols: [
+        'OpenVPN'
+      ]
+      lock: {
+        name: 'myCustomLockName'
+        kind: 'CanNotDelete'
+      }
```
Comment (AlexanderSehr) [inline avm/res/network/vpn-server-configuration/tests/e2e/waf-aligned/main.test.bicep:84]:
```suggestion
```

not required for WAF-alignement. Please note that this change requires the readme to be regenerated

File: avm/res/network/vpn-server-configuration/tests/e2e/defaults/main.test.bicep [hunk 3]
```diff
@@ -0,0 +1,84 @@
...
+
+@batchSize(1)
+module testDeployment '../../../main.bicep' = [
+  for iteration in ['init', 'idem']: {
+    scope: resourceGroup
+    name: '${uniqueString(deployment().name, resourceLocation)}-test-${serviceShort}-${iteration}'
+    params: {
+      location: resourceLocation
+      name: '${namePrefix}${serviceShort}VPNConfig'
+      aadAudience: '11111111-1234-4321-1234-111111111111'
```
Comment (AlexanderSehr) [inline avm/res/network/vpn-server-configuration/tests/e2e/defaults/main.test.bicep:59]: Please note that this test should only contain the minimum set of parameters required to deploy. From the looks of it, all parameters but the name (& and maybe location) are optional. Are they required afterall?

File: avm/res/network/vpn-server-configuration/tests/e2e/defaults/main.test.bicep (hunk 1 above)
Comment (ericscheffler) [reply avm/res/network/vpn-server-configuration/tests/e2e/defaults/main.test.bicep:12]: <sigh> I guess....

Also, question about line #199 in the main.bicep file I marked the user-defined type as exportable (I plagiarized that from another unrelated resource example); while I'm at it should I remove this as well?

File: avm/res/network/vpn-server-configuration/tests/e2e/defaults/main.test.bicep (hunk 3 above)
Comment (ericscheffler) [reply avm/res/network/vpn-server-configuration/tests/e2e/defaults/main.test.bicep:59]: So, in order for the resource to deploy it has to have _at least_ one identity source configured (AAD, RADIUS or cert); I figured that AAD would be the most common, so I included that one here.

Comment (AlexanderSehr) [reply avm/res/network/vpn-server-configuration/tests/e2e/defaults/main.test.bicep:59]: Sounds good. Thanks for elaborating :)

File: avm/res/network/vpn-server-configuration/tests/e2e/defaults/main.test.bicep (hunk 1 above)
Comment (AlexanderSehr) [reply avm/res/network/vpn-server-configuration/tests/e2e/defaults/main.test.bicep:12]: No, the `export()` is fine. We may actually do this for all UDTs in the `main.bicep` templates eventually to enable users to import & re-use those types for their own parameters 💪

**Comment by ReneHezser** [comment]
@ericscheffler 
We have added examples for Bicep parameter files to the Readme. This has been applied to all published modules but needs to be done for PRs as well. Can you please run the Set-AVMModule utility as detailed [here](https://azure.github.io/Azure-Verified-Modules/contributing/bicep/bicep-contribution-flow/generate-bicep-module-files/). It is required for the validation pipeline to succeed and the contribution to be published.

Please reach out if any support is needed.

**Review by AlexanderSehr — APPROVED** [review]

**Pull request merged**